t/043-variable-access-through-expansion.t.cpp

```diff
@@ -0,0 +1,100 @@
+#include "clangmetatool-testconfig.h"
+
+#include <gtest/gtest.h>
+
+#include <clangmetatool/meta_tool_factory.h>
+#include <clangmetatool/meta_tool.h>
+#include <clangmetatool/collectors/include_graph.h>
+
+#include <clang/Frontend/FrontendAction.h>
+#include <clang/Tooling/Core/Replacement.h>
+#include <clang/Tooling/CommonOptionsParser.h>
+#include <clang/Tooling/Tooling.h>
+#include <clang/Tooling/Refactoring.h>
+#include <llvm/Support/CommandLine.h>
+
+class MyTool {
+private:
+  clang::CompilerInstance* ci;
+  clangmetatool::collectors::IncludeGraph graph;
+public:
+  MyTool(clang::CompilerInstance* ci, clang::ast_matchers::MatchFinder *f)
+    :ci(ci), graph(ci, f) {
+  }
+  void postProcessing
+  (std::map<std::string, clang::tooling::Replacements> &replacementsMap) {
+    clangmetatool::collectors::IncludeGraphData *data = graph.getData();
+
+    // file ID 0 to 1, aka foo.cpp to paste.h
+    auto edge = std::make_pair(0, 1);
+    ASSERT_EQ(data->usage_reference_count.count(edge), 1);
+    ASSERT_EQ(data->usage_reference_count[edge], 2); // reference PASTE 2x
+
+    // file ID from 0 to 2, aka foo.cpp to global.h
+    edge = std::make_pair(0, 2);
+    ASSERT_EQ(data->usage_reference_count.count(edge), 1);
+    // GLOBAL1 .. GLOBAL4 are referenced
+    ASSERT_EQ(data->usage_reference_count[edge], 4);
+
+    // file ID from 2 to 1, aka paste.h to global.h
+    edge = std::make_pair(1, 2);
+    ASSERT_EQ(data->usage_reference_count.count(edge), 1);
+    ASSERT_EQ(data->usage_reference_count[edge], 0);
+
+    // file ID from 0 to 3, aka foo.cpp to macro.h
+    edge = std::make_pair(0, 3);
+    ASSERT_EQ(data->usage_reference_count[edge], 2); // USE 2 macros from here
+
+    // file ID from 3 to 2, aka macro.h to global.h
+    edge = std::make_pair(3, 2);
+    // Even though macro.h has a "spelling" of GLOBAL3 in macro.h it does
+    //   not include global.h and does not directly reference GLOBAL3
+    //   only callers of the macro actually reference global.h
```

Comment:
What if it did include global.h? Would that change things? Seems like the reference is really macro.h -> global.h in that case, and maybe in the case in the test too but we are just getting the include transitively.

Reply:
When that we add `global.h` to `macro.h` after this change we have
```c++
data->usage_reference_count.count(edge) == 1;
data->usage_reference_count[edge] == 0;
```

So the edge exists but there are no references from `macro.h` to `global.h`.

Reply:
Will address that corner case later once we have a solid use case that requires handling it.